core/ustring.cpp

```diff
@@ -2957,23 +2957,20 @@ String String::replace(const char *p_key, const char *p_with) const {
 String String::replace_first(const String &p_key, const String &p_with) const {
 
 	String new_string;
-	int search_from = 0;
 	int result = 0;
 
-	while ((result = find(p_key, search_from)) >= 0) {
+	if ((result = find(p_key, 0)) >= 0) {
```

Comment:
That's still too convoluted, this code could simply be something like that (untested):
```cpp
int pos = find(p_key);
if (pos >= 0) {
	return substr(0, pos) + p_with + substr(pos + p_key.length(), length());
} else {
	return *this;
}
```